pkg/types/etcdconnection.go

```diff
@@ -69,6 +70,7 @@ func NewEtcdConnectionConfig() *EtcdConnectionConfig {
 // AddFlags adds the flags to flagset.
 func (c *EtcdConnectionConfig) AddFlags(fs *flag.FlagSet) {
 	fs.StringSliceVarP(&c.Endpoints, "endpoints", "e", c.Endpoints, "comma separated list of etcd endpoints")
+	fs.StringSliceVar(&c.ServiceEndpoints, "service-endpoints", c.ServiceEndpoints, "comma separated list of etcd endpoints that are used for etcd-backup-restore to connect to etcd through a (Kubernetes) service")
```

Comment:
One doubt:
What will be its default value, what if I want to run backup-restore as a process not in k8s ... will it cause any issue?

Reply:
The default value will be what is passed to `--endpoints`. If you don't run it as part of K8s then you need to take care of providing an endpoint that forwards traffic only to those members which are ready to service traffic. This is how the scale-up feature was designed and built from the beginning.

Reply:
How it will be behave in case replica=1 ?

Reply:
The replica count shouldn't matter here. Also with the earlier implementation when the single replica was not running then the backup-restore side-car could not connect to etcd. Do you have doubts?